### PR TITLE
Update CHANGELOG for 9.5.0

### PR DESCRIPTION
**Pull request by onurctirtir**
(will squash commits before merging, split for easier review)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,103 @@
...
+
+* Adds support for `EXPLAIN ANALYZE EXECUTE` and `EXPLAIN EXECUTE`
+
+* Adds support for `EXPLAIN(ANALYZE, WAL)` for PG13
+
+* Adds support for `WITH TIES` option in SELECT and INSERT SELECT queries
+
+* Avoids taking multi-shard locks on workers
+
+* COPY honors `citus.max_shared_pool_size` config
```
Comment (hanefi) [inline CHANGELOG.md:19]: All the other entries start with a verb. How about:

```suggestion
* Enforces `citus.max_shared_pool_size` config in COPY queries
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,103 @@
...
+
+* Fixes `citus_finish_pg_upgrade` to drop all backup tables
+
+* Fixes a bug that cause failures when `RECURSIVE VIEW` joined reference table
+
+* Fixes DROP SEQUENCE failures when metadata syncing is enabled
+
+* Fixes a bug that caused CREATE TABLE with CHECK constraint to fail
+
+* Fixes a bug that caused VACUUM to deadlock in PG13
```
Comment (JelteF) [inline CHANGELOG.md:53]: Seems irrelevant since we introduced PG13 support here

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,103 @@
...
+
+* Fixes write queries with const expressions and COLLATE in various places
+
+* Fixes wrong cancellation message about distributed deadlock
+
+* Forces aliases in deparsing queries with anonymous column references
+
+* Improves error message when creating a foreign key to a local table
+
+* Makes `citus_prepare_pg_upgrade` idempotent by dropping transition tables
```
Comment (JelteF) [inline CHANGELOG.md:99]: I would move this one up in the list, somewhere before all the various fixes.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,103 @@
...
+
+* Prevents pull-push execution for simple pushdownable subqueries
+
+* Sorts the output of `EXPLAIN ANALYZE` by execution duration.
+
+* Disallows `ON TRUE` outer joins with ref & dist tables when ref table is outer
+
+* Disallows field indirection in INSERT/UPDATE queries
+
+* Disallows volatile functions in UPDATE subqueries
```
Comment (JelteF) [inline CHANGELOG.md:37]: Maybe make clear that these items are because they crashed or returned bad results. Now it reads like we're simply removing features.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,103 @@
...
+
+* Enforces `citus.max_intermediate_result_size` in local execution
+
+* Improves cost estimation of INSERT SELECT plans
+
+* Introduces delegation of procedures that read from reference tables
+
+* Prevents pull-push execution for simple pushdownable subqueries
+
+* Sorts the output of `EXPLAIN ANALYZE` by execution duration.
```
Comment (JelteF) [inline CHANGELOG.md:31]: I would put this together with the other EXPLAIN ANALYZE improvements.

File: CHANGELOG.md (hunk 2 above)
Comment (onurctirtir) [reply CHANGELOG.md:53]: That's true. But if I understood correctly, @marcocitus thought it might worth to mention as this is also changing the behavior for pg11&12 as well. Removing seems to make more sense, what do you think @marcocitus @JelteF  ?

File: CHANGELOG.md (hunk 3 above)
Comment (onurctirtir) [reply CHANGELOG.md:99]: which one, Improves.. or Makes.. ?

File: CHANGELOG.md (hunk 5 above)
Comment (onurctirtir) [reply CHANGELOG.md:31]: makes sense

File: CHANGELOG.md (hunk 4 above)
Comment (onurctirtir) [reply CHANGELOG.md:37]: What about adding ` to avoid incorrect results` to the end of each one ?

File: CHANGELOG.md (hunk 2 above)
Comment (JelteF) [reply CHANGELOG.md:53]: If it's reworded to also apply to pg11 & pg12 I think it makes sense to keep.

File: CHANGELOG.md (hunk 3 above)
Comment (JelteF) [reply CHANGELOG.md:99]: I meant "Makes ...", but the improves makes sense to move as well.

File: CHANGELOG.md (hunk 5 above)
Comment (onurctirtir) [reply CHANGELOG.md]: done 0f5c6d276e0d5c3d0ad562ecb9f9bbab6e946e0b

File: CHANGELOG.md (hunk 4 above)
Comment (onurctirtir) [reply CHANGELOG.md]: done at 2ba6477947332e2ec6c4fbf5ea361ff4f4c80861

File: CHANGELOG.md (hunk 3 above)
Comment (onurctirtir) [reply CHANGELOG.md]: I think both, done at a5f62e850cd228a3dbab8c8c6d32f38b611a58a7

File: CHANGELOG.md (hunk 2 above)
Comment (onurctirtir) [reply CHANGELOG.md]: then rewording it to `Fixes a bug that could cause VACUUM to deadlock` at 9ae6e4d34ddb8008f7356671cf0c5dd17cc171a5

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,105 @@
...
+
+* Fixes a crash when aggregating empty tables
+
+* Fixes a crash with inserting domain constrained composite types
+
+* Fixes a crash with multi-row & router INSERT's in local execution
+
+* Fixes a memory leak in subtransaction memory handling
+
+* Fixes a possibility of doing cleanup more than once
```
Comment (marcocitus) [inline CHANGELOG.md:85]: What's cleaned up?

**Review by marcocitus — APPROVED** [review]

File: CHANGELOG.md (hunk 6 above)
Comment (onurctirtir) [reply CHANGELOG.md]: addressed

**Review by hanefi — APPROVED** [review]

**Comment by onurctirtir** [comment]
@metdos noticed, we should add https://github.com/citusdata/citus/pull/4250 as well
